crates/npmrc/src/custom_deserializer.rs

```diff
@@ -16,30 +16,40 @@ pub fn default_public_hoist_pattern() -> Vec<String> {
     vec!["*eslint*".to_string(), "*prettier*".to_string()]
 }
 
+/// Get absolute path with home_dir
+pub fn get_absolute_path_with_home_dir(relate_path: Option<String>) -> PathBuf {
+    let home_dir = dirs::home_dir().expect("Home directory is not available");
+    match relate_path {
+        Some(p) => {
+            // prefix should start with ~/
+            if let Some(path_without_tilde) = p.strip_prefix("~/") {
+                return home_dir.join(path_without_tilde);
+            }
+            PathBuf::from(p)
+        }
+        None => home_dir,
+    }
+}
+
```

Comment:
I don't think this function is necessary.

You can replace all occurrences of `get_absolute_path_with_home_dir(None)` with just `dirs::home_dir()`.

For the 2 occurrences of `get_absolute_path_with_home_dir(Some(_))`, you can replace it with expressions then refactor a bit.